analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/BranchingProcessor.cs

```diff
@@ -41,7 +41,12 @@ protected override ProgramState[] Process(SymbolicContext context, T operation)
         var state = PreProcess(context.State, operation);
         if (BoolConstraintFromOperation(state, operation, context.IsLoopCondition, context.VisitCount) is { } constraint)
         {
-            return state.SetOperationConstraint(context.Operation, constraint).ToArray();    // We already know the answer from existing constraints
+            state = state.SetOperationConstraint(context.Operation, constraint);    // We already know the answer from existing constraints
+            if (context.VisitCount > 1)
```

Comment:
What other processor except Binary benefits from this?

Reply:
I don't think any. Refactoring this logic into `Binary` will be very tricky, though.